Modules/01_oop_and_ood/src/Point.ts

```diff
@@ -0,0 +1,33 @@
+export class Point {
+  x: number;
+  y: number;
+ 
+  constructor(x?: number, y?: number){
+    this.x = x || 0;
+    this.y = y || 0;
+  }
+
+  toString(): string {
+    return `(${this.x}, ${this.y})`
+  }
+
+  distance(): number;
+  distance(point: Point): number;
+  distance(x: number, y: number): number;
+
+  distance(paramOne?: number | Point, paramTwo?: number): number {
+    if (!paramOne) {
+      return this.getDistance();
+    } else if (paramOne instanceof Point && !paramTwo) {
```

Comment:
I do not agree, since if you would pass `Point` and `number` params and expect that they both would be used - it would be a mistake, since the second param would be useless. Also it's an edge case and there were no rules for this. But it definetly has a purpose ;)

Reply:
Alright, but do you have an overload which accepts a Point and a number as a second parameter? Of course you can ignore the ts errors, but this would just be trying to cause errors deliberately.